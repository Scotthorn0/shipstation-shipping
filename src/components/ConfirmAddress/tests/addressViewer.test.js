import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'semantic-ui-react';
import AddressViewer from '../addressViewer';

const address = {
  name: 'MICKEY AND MINNIE MOUSE',
  phone: '714-781-4565',
  company_name: 'THE WALT DISNEY COMPANY',
  address_line1: '500 S BUENA VISTA ST',
  address_line2: '',
  address_line3: null,
  city_locality: 'BURBANK',
  state_province: 'CA',
  postal_code: '91521-0007',
  country_code: 'US',
  address_residential_indicator: 'no',
};

describe.only('AddressViewer Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddressViewer address={address} />);
  });

  test('should render without crashing', () => {
    expect(wrapper.exists()).toBe.true;
  });

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should not render lines for falsy values', () => {
    expect(wrapper.children()).toHaveLength(8);
    const updatedAddress = { ...address, address_line2: 'Unit# 4000' };
    wrapper.setProps({ address: updatedAddress });
    expect(wrapper.children()).toHaveLength(9);
  });

  test('should not show the residential indicator field', () => {
    expect(wrapper.children().last().contains(<List.Item>US</List.Item>)).toBe.true;
    expect(wrapper.children().last().contains(<List.Item>no</List.Item>)).toBe.false;
  });
});
