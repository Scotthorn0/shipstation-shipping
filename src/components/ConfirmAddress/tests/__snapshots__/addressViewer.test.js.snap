// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressViewer Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressViewer
    address={
        Object {
            "address_line1": "500 S BUENA VISTA ST",
            "address_line2": "",
            "address_line3": null,
            "address_residential_indicator": "no",
            "city_locality": "BURBANK",
            "company_name": "THE WALT DISNEY COMPANY",
            "country_code": "US",
            "name": "MICKEY AND MINNIE MOUSE",
            "phone": "714-781-4565",
            "postal_code": "91521-0007",
            "state_province": "CA",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ListItem>
          MICKEY AND MINNIE MOUSE
</ListItem>,
        <ListItem>
          714-781-4565
</ListItem>,
        <ListItem>
          THE WALT DISNEY COMPANY
</ListItem>,
        <ListItem>
          500 S BUENA VISTA ST
</ListItem>,
        null,
        null,
        <ListItem>
          BURBANK
</ListItem>,
        <ListItem>
          CA
</ListItem>,
        <ListItem>
          91521-0007
</ListItem>,
        <ListItem>
          US
</ListItem>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "name",
        "nodeType": "class",
        "props": Object {
          "children": "MICKEY AND MINNIE MOUSE",
        },
        "ref": null,
        "rendered": "MICKEY AND MINNIE MOUSE",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "phone",
        "nodeType": "class",
        "props": Object {
          "children": "714-781-4565",
        },
        "ref": null,
        "rendered": "714-781-4565",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "company_name",
        "nodeType": "class",
        "props": Object {
          "children": "THE WALT DISNEY COMPANY",
        },
        "ref": null,
        "rendered": "THE WALT DISNEY COMPANY",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "address_line1",
        "nodeType": "class",
        "props": Object {
          "children": "500 S BUENA VISTA ST",
        },
        "ref": null,
        "rendered": "500 S BUENA VISTA ST",
        "type": [Function],
      },
      null,
      null,
      Object {
        "instance": null,
        "key": "city_locality",
        "nodeType": "class",
        "props": Object {
          "children": "BURBANK",
        },
        "ref": null,
        "rendered": "BURBANK",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "state_province",
        "nodeType": "class",
        "props": Object {
          "children": "CA",
        },
        "ref": null,
        "rendered": "CA",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "postal_code",
        "nodeType": "class",
        "props": Object {
          "children": "91521-0007",
        },
        "ref": null,
        "rendered": "91521-0007",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "country_code",
        "nodeType": "class",
        "props": Object {
          "children": "US",
        },
        "ref": null,
        "rendered": "US",
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ListItem>
            MICKEY AND MINNIE MOUSE
</ListItem>,
          <ListItem>
            714-781-4565
</ListItem>,
          <ListItem>
            THE WALT DISNEY COMPANY
</ListItem>,
          <ListItem>
            500 S BUENA VISTA ST
</ListItem>,
          null,
          null,
          <ListItem>
            BURBANK
</ListItem>,
          <ListItem>
            CA
</ListItem>,
          <ListItem>
            91521-0007
</ListItem>,
          <ListItem>
            US
</ListItem>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "name",
          "nodeType": "class",
          "props": Object {
            "children": "MICKEY AND MINNIE MOUSE",
          },
          "ref": null,
          "rendered": "MICKEY AND MINNIE MOUSE",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "phone",
          "nodeType": "class",
          "props": Object {
            "children": "714-781-4565",
          },
          "ref": null,
          "rendered": "714-781-4565",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "company_name",
          "nodeType": "class",
          "props": Object {
            "children": "THE WALT DISNEY COMPANY",
          },
          "ref": null,
          "rendered": "THE WALT DISNEY COMPANY",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "address_line1",
          "nodeType": "class",
          "props": Object {
            "children": "500 S BUENA VISTA ST",
          },
          "ref": null,
          "rendered": "500 S BUENA VISTA ST",
          "type": [Function],
        },
        null,
        null,
        Object {
          "instance": null,
          "key": "city_locality",
          "nodeType": "class",
          "props": Object {
            "children": "BURBANK",
          },
          "ref": null,
          "rendered": "BURBANK",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "state_province",
          "nodeType": "class",
          "props": Object {
            "children": "CA",
          },
          "ref": null,
          "rendered": "CA",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "postal_code",
          "nodeType": "class",
          "props": Object {
            "children": "91521-0007",
          },
          "ref": null,
          "rendered": "91521-0007",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "country_code",
          "nodeType": "class",
          "props": Object {
            "children": "US",
          },
          "ref": null,
          "rendered": "US",
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
